pkg/menu/menu.go

```diff
@@ -49,25 +81,120 @@ func processInput(input, warning *widgets.Paragraph, isValid func(string) (strin
+// NewInputWindow opens a new input window with fixed width=100, hight=1.
+func NewInputWindow(introwords string, isValid func(string) (string, string, bool)) (string, error) {
+	uiEvents := ui.PollEvents()
+	inputString, err := newInputWindow(introwords, 100, 1, isValid, uiEvents)
+	return inputString, err
+}
+
+func newInputWindow(introwords string, wid int, ht int, isValid func(string) (string, string, bool), uiEvents <-chan ui.Event) (string, error) {
+	if err := ui.Init(); err != nil {
+		return "", fmt.Errorf("Failed to initialize termui: %v", err)
+	}
+	defer ui.Close()
 
+	input, _, err := processInput(introwords, 0, wid, ht, isValid, uiEvents)
+
+	return input, err
+}
+
+// DisplayResult opens a new window and displays a message.
+// each item in the message array will be displayed on a single line.
+func DisplayResult(message []string, wid int) (string, error) {
+	uiEvents := ui.PollEvents()
+	return displayResult(message, wid, uiEvents)
+}
+
+func displayResult(message []string, wid int, uiEvents <-chan ui.Event) (string, error) {
+	if err := ui.Init(); err != nil {
+		return "", fmt.Errorf("Failed to initialize termui: %v", err)
+	}
+	defer ui.Close()
+
+	p := widgets.NewParagraph()
+	p.Text = strings.Join(message, "\n")
+	p.Border = false
+	p.SetRect(0, 0, wid, len(message)+3)
+	p.TextStyle.Fg = ui.ColorWhite
+
+	ui.Render(p)
+
+	// press any key to exit this window
+	k := readKey(uiEvents)
+	if k == "<C-d>" {
+		return p.Text, io.EOF
+	}
+	return p.Text, nil
+}
+
+// DisplayMenu presents all entries into a menu with numbers.
+// user inputs a number to choose from them.
+func DisplayMenu(menuTitle string, introwords string, entries []Entry) (Entry, error) {
 	uiEvents := ui.PollEvents()
-	return processInput(input, warning, isValid, uiEvents)
+	return displayMenu(menuTitle, introwords, entries, uiEvents)
+}
+
+func displayMenu(menuTitle string, introwords string, entries []Entry, uiEvents <-chan ui.Event) (Entry, error) {
+	if err := ui.Init(); err != nil {
+		return nil, fmt.Errorf("Failed to initialize termui: %v", err)
+	}
+	defer ui.Close()
+	// listData contains all choice's labels
+	listData := []string{}
+
+	for i, e := range entries {
+		listData = append(listData, fmt.Sprintf("[%d] %s", i, e.Label()))
+	}
+
+	location := 0
+	l := widgets.NewList()
+	l.Title = menuTitle
+	l.Rows = listData
+	l.SetRect(0, location, menuWidth, location+len(entries)+2)
+	location += len(entries) + 2
+	l.TextStyle.Fg = ui.ColorWhite
+	ui.Render(l)
+
+	// we want user to choose from the menu, so the isValid will check :
+	// 1.input is a number; 2.input number does not exceed the number of options.
+	isValid := func(input string) (string, string, bool) {
+		if input == "" {
+			for i, en := range entries {
+				if en.IsDefault() {
+					return strconv.Itoa(i), "", true
+				}
+			}
+			return "-1", "No default option, please enter a choice", false
+		}
+		if c, err := strconv.ParseUint(input, 10, 32); err != nil || c > uint64(len(entries)) {
```

Comment:
What about using strconv.Atoi here and below? https://golang.org/pkg/strconv/#Atoi